src/main/kotlin/Menu.kt

```diff
@@ -0,0 +1,76 @@
+class Menu<T>(private val data: T) {
+    private var maxPunkt: Int = 0
+    fun show() {
+        while (true) {
+            var i = 0
+
+            if (data is Storage) {
+                maxPunkt = data.body.size + 1
+                println("Список архивов:\n0. Создать архив")
+
+                data.body.forEach {
+                    i++
+                    println("$i. ${it.name}")
+                }
+            }
+            if (data is Archive) {
+                maxPunkt = data.body.size + 1
+                println("Список заметок:\n0. Создать заметку")
+                data.body.forEach {
+                    i++
+                    println("$i. ${it.name}")
+                }
+            }
+            if (data is Note) {
+                maxPunkt = 0
+                println(data)
+            }
+            println("$maxPunkt. Выход")
+
+            println("\nВведите номер команды:")
+
+            val stringCommand = readlnOrNull() ?: ""
+            var command: Int = -1
+            try {
+                command = stringCommand.toInt()
+            } catch (e: Exception) {
+                println("Ошибка: введите число!")
+                continue
+            }
+
+            if (command in 0..maxPunkt)
+                when (command) {
```

Comment:
🍏 Можно немного уменьшить вложенность:
```suggestion
                when {
                command == maxPunkt -> {...} 
                command == 0 -> {...}
                data is Storage -> {...}
                data is Archive -> {...}
                else -> Unit
                }
                
```